build-package.sh

```diff
@@ -169,16 +169,22 @@ if [ "$(uname)" == "Darwin" ]; then
 fi
 
 # Generate jazzy docs (macOS) where the 'documentation' tag exists on the issue label
-if [ "$(uname)" == "darwin"* ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
-    if  [ -z "${GITHUB_USERNAME}" ] && [ -z "${GITHUB_PASSWORD}" ]; then
-
+if [ "$(uname)" == "Darwin" ] && [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
+    if  [ -n "${GITHUB_USERNAME}" -a -n "${GITHUB_PASSWORD}" ]; then
+        echo "Checking PR for docs generation tag"
         # Obtain the name of the tag from the GitHub repo using the GitHub Username and Password, discarding unnecessary text around the retrieved string
-        response= curl -s -X GET echo "https://${GITHUB_USERNAME}:${GITHUB_PASSWORD}@api.github.com/repos/${TRAVIS_REPO_SLUG}/issues/${TRAVIS_PULL_REQUEST}/labels" | sed 's/\\\\\//\//g' | sed 's/[{}]//g' | awk -v k="text" '{n=split($0,a,","); for (i=1; i<=n; i++) print a[i]}' | sed 's/\"\:\"/\|/g' | sed 's/[\,]/ /g' | sed 's/\"//g' | grep -w 'name'
-
+        jsonResponse=`curl -s -X GET https://${GITHUB_USERNAME}:${GITHUB_PASSWORD}@api.github.com/repos/${TRAVIS_REPO_SLUG}/issues/${TRAVIS_PULL_REQUEST}/labels`
+        echo "Label data retreived: $jsonResponse"
+        labelNames=`echo "$jsonResponse" | grep '"name"'`
+        # Extract the label name from the "name": "value" pair. This assumes each pair is on a separate line.
+        candidateTags=`echo "$labelNames" | sed -e's#.*"name" *: *"\([^"]*\)".*#\1#'`
+        echo "Labels: " $candidateTags
```

Comment:
The above is split into stages, so that we can easily debug when the response from the Github API is not what we expect.  I also simplified the extraction of label names into a single `grep` and `sed` as I couldn't understand the existing algorithm (although, it did appear to work).
A caveat is that this assumes the JSON returned by Github will produce entries on separate lines (which it currently does).  I think this is a safe assumption for now.